tests/test_gsobject_model.py

```diff
@@ -0,0 +1,483 @@
+
+        # Now try fastfit=False.
+        print('Slow fit')
+        model = piff.GSObjectModel(fiducial, fastfit=False)
+        fit = model.fit(star).fit
+
+        print('True size = ',size,', model size = ',fit.params[0])
+        print('True g1 = ',g1,', model g1 = ',fit.params[1])
+        print('True g2 = ',g2,', model g2 = ',fit.params[2])
+        print('True cenu = ', cenu, ', model cenu = ', fit.center[0])
+        print('True cenv = ', cenv, ', model cenv = ', fit.center[1])
+
+        # The sensitivity of this test is somewhat dependent on what the wcs is, and whether the fit
+        # center parameters are in image coords or world coords.
+        np.testing.assert_allclose(fit.params[0], size, rtol=1e-3)
+        np.testing.assert_allclose(fit.params[1], g1, atol=1e-3)
+        np.testing.assert_allclose(fit.params[2], g2, atol=1e-3)
+        np.testing.assert_allclose(fit.center[0], cenu, atol=1e-3)
+        np.testing.assert_allclose(fit.center[1], cenv, atol=1e-3)
+
+        # # Now test running it via the config parser
+        # config = {
+        #     'model' : {
+        #         'type' : 'GSObjectModel'
+        #     }
+        # }
+        # if __name__ == '__main__':
+        #     logger = piff.config.setup_logger(verbose=3)
+        # else:
+        #     logger = piff.config.setup_logger(verbose=1)
+        # model = piff.Model.process(config['model'], logger)
+        # fit = model.fit(star).fit
+        #
+        # # Same tests.
+        # np.testing.assert_allclose(fit.params[0], size, rtol=1e-3)
+        # np.testing.assert_allclose(fit.params[1], g1, atol=1e-3)
+        # np.testing.assert_allclose(fit.params[2], g2, atol=1e-3)
+
+        # Also need to test ability to serialize
+        outfile = os.path.join('output', 'gsobject_test.fits')
+        with fitsio.FITS(outfile, 'rw', clobber=True) as f:
+            model.write(f, 'psf_model')
+        with fitsio.FITS(outfile, 'r') as f:
+            roundtrip_model = piff.GSObjectModel.read(f, 'psf_model')
+        assert model.__dict__ == roundtrip_model.__dict__
+
+def test_center():
+    """Fit with centroid free and PSF center constrained to an initially mis-registered PSF.
+    """
+    influx = 150.
+    size = 2.0
+    u0, v0 = 0.6, -0.4
+    g1, g2 = 0.1, 0.2
+    for fiducial in [fiducial_gaussian, fiducial_kolmogorov, fiducial_moffat]:
+        s = make_data(fiducial, size, g1, g2, u0, v0, influx, du=0.5)
+
+        mod = piff.GSObjectModel(fiducial)
+        star = mod.initialize(s)
+        print('Flux, ctr after reflux:',star.fit.flux,star.fit.center)
+        for i in range(3):
+            star = mod.fit(star)
+            star = mod.reflux(star)
+            print('Flux, ctr, chisq after fit {:d}:'.format(i),
+                  star.fit.flux, star.fit.center, star.fit.chisq)
+            # These fluxes are not at all close to influx.  Not sure why...
+            # np.testing.assert_almost_equal(star.fit.flux/influx, 1.0, decimal=2)
```

Comment:
Just kidding.  These pass at 14 digits of accuracy, not 2.

I went through and upped the test requirements throughout this file.